File: core/src/main/java/com/egoal/darkestpixeldungeon/items/weapon/melee/BattleGloves.java
package com.egoal.darkestpixeldungeon.items.weapon.melee;

import com.egoal.darkestpixeldungeon.sprites.ItemSpriteSheet;

/**
 * Created by 93942 on 8/16/2018.
 */

public class BattleGloves extends MeleeWeapon {

  {
    image = ItemSpriteSheet.DPD_BATTLE_GLOVES;

    tier = 1;
    DLY = .75f;  // faster speed
    // ACC		=	1.2f;	// 20% boost to accuracy
  }

  // 3+2*x
  @Override
  public int max(int lvl) {
    // very low damage
    return 3 * (tier) + lvl * (tier + 1);
  }

  @Override
  public int STRReq(int lvl) {
    lvl = Math.max(0, lvl);
    return (6 + tier * 2) - (int) (Math.sqrt(8 * lvl + 1) - 1) / 2;
  }
}

File: core/src/main/java/com/egoal/darkestpixeldungeon/items/weapon/melee/BattleGloves.kt
package com.egoal.darkestpixeldungeon.items.weapon.melee

import com.egoal.darkestpixeldungeon.actors.hero.Hero
import com.egoal.darkestpixeldungeon.sprites.ItemSpriteSheet
import kotlin.math.pow

/**
 * Created by 93942 on 8/16/2018.
 */

class BattleGloves : MeleeWeapon() {

    init {
        image = ItemSpriteSheet.DPD_BATTLE_GLOVES

        tier = 1
        DLY = 0.85f  // faster speed
        // ACC		=	1.2f;	// 20% boost to accuracy
    }

    // 3+2*x, lower base damage
    override fun max(lvl: Int): Int = 3 * tier + lvl * (tier + 1)

    override fun STRReq(lvl: Int): Int = super.STRReq(lvl) - 2

    override fun speedFactor(hero: Hero): Float = super.speedFactor(hero) * 0.9f.pow(level())
}
